File: app/src/main/java/org/tvheadend/tvhclient/di/modules/SharedPreferencesModule.java
package org.tvheadend.tvhclient.di.modules;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import androidx.annotation.NonNull;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class SharedPreferencesModule {

    @Provides
    @NonNull
    @Singleton
    SharedPreferences provideSharedPreferences(Context context) {
        return PreferenceManager.getDefaultSharedPreferences(context);
    }
}

File: app/src/main/java/org/tvheadend/tvhclient/di/module/SharedPreferencesModule.kt
package org.tvheadend.tvhclient.di.module

import android.content.Context
import android.content.SharedPreferences
import androidx.preference.PreferenceManager
import dagger.Module
import dagger.Provides
import org.tvheadend.data.di.FeatureScope

@Suppress("unused")
@Module
class SharedPreferencesModule {

    @Provides
    @FeatureScope
    internal fun provideSharedPreferences(context: Context): SharedPreferences {
        return PreferenceManager.getDefaultSharedPreferences(context)
    }
}
